docs/writing-tests/test-runner-with-vitest.mdx

```diff
@@ -0,0 +1,396 @@
+- [`storybookScript`](#storybookscript): Ensure this script is correctly starting Storybook.
+
+If the URLs are not working when running tests in CI, you should ensure the Storybook is built and published before running the tests. You can then provide the URL to the published Storybook using the `storybookUrl` option. See the [In CI](#in-ci) section for an example.
+
+### How to apply custom Vite configuration?
+
+If you have custom operations defined in [`viteFinal`](../api/main-config/main-config-vite-final.mdx) in your `.storybook/main.js|ts` file, you will need to translate those into the Vitest configuration. This is because the plugin does not use the Storybook Vite configuration.
+
+```ts
+TK - Is there a good example we could offer here?
+```
+
+### Why do we recommend browser mode?
+
+Vitest's browser mode runs your tests in a real browser (Chromium, via Playwright, in the default configuration). The alternative is a simulated browser environment, like JSDom or HappyDom, which can have differences in behavior compared to a real browser. For UI components, which can often depend on browser APIs or features, running tests in a real browser is more accurate.
+
+For more, see [Vitest's guide on using browser mode effectively](https://vitest.dev/guide/browser/#motivation).
+
+### How to use WebDriver instead of Playwright?
+
+We recommend running tests in a browser using Playwright, but you can use WebDriverIO instead. To do so, you need to adjust the [browser provider in the Vitest configuration file](https://vitest.dev/config/#browser-provider).
+
+### How to use a browser other than Chromium
+
+We recommend using Chromium, because it is most likely to best match the experience of a majority of your users. However, you can use other browsers by adjusting the [browser name in the Vitest configuration file](https://vitest.dev/config/#browser-name). Note that [Playwright and WebDriverIO support different browsers](https://vitest.dev/guide/browser/#browser-option-types).
+
+### How is this different from the previous test runner?
+
+The [previous test runner](./test-runner.mdx) requires a running Storybook instance to test your stories, because it visits each one, executes the play function, and listens for results. This plugin, however, transforms your stories into tests using Vite and portable stories, so it does not need to run Storybook to test your stories. Beyond that core difference, there are a few other distinctions:
+
+Additionally, the previous test runner ran your stories as orchestrated tests in Jest, and that orchestration came with some complexity. By comparison, this plugin transforms your stories into real tests and then runs them using Vitest, which is simpler and more configurable.
+
+Finally, because of the simpler architecture and the use of Vitest, this plugin should be faster than the previous test runner for most projects. We'll do more benchmarking to quantify this in the future.
```

Comment:
Sorry, I'm unclear on what you're asking for:

> Move this up...

What specifically is _this_ here? The whole section? Just the last paragraph?

> before the FAQ item

I think understanding what _this_ refers to will help, but I'm also not sure how to interpret this one

> we should avoid calling the original test runner as the previous one

Why? And how else should I refer to it?

Reply:
What I was referring to was removing this from the FAQ items, making it a section of itself, placing it above said section, and turning it into something as follows (based on the current content structure)

```md

### Debugging

While the plugin does not require Storybook to run when testing, you may still want to run Storybook to debug your tests......

****

### How is this different from the previous test runner?

The [previous test runner](./test-runner.mdx) requires a running Storybook instance to test your stories because it visits each one, executes the play function, and listens for results. This plugin, however, transforms your stories into tests using Vite and portable stories, so it does not need to run Storybook to test your stories. Beyond that core difference, there are a few other distinctions:

Additionally, the previous test runner ran your stories as orchestrated tests in Jest, and that orchestration came with some complexity. By comparison, this plugin transforms your stories into real tests and then runs them using Vitest, which is simpler and more configurable.

Finally, because of the simpler architecture and the use of Vitest, this plugin should be faster than the previous test runner for most projects. We'll do more benchmarking to quantify this in the future.

## FAQ

```



Applying this small change to the documentation increases the feature's discoverability and makes it easier for users to understand its purpose, key benefits, and why they should use it. As how it's currently documented, it's buried deep in the documentation (i.e., it's the last item on the FAQ).

Reply:
Thanks for the additional details. I agree with this suggestion.